docs/kernel/modify_kernel.md

```diff
@@ -2,105 +2,182 @@
+"+ cat config_diff"
+"--- new_config\t2023-04-07 04:31:15.263160807 +0000"
+"+++ current_config\t2023-04-07 04:31:12.255183015 +0000"
+# ...
+"@@ -862,7 +862,6 @@"
+" CONFIG_BLK_DEV_THROTTLING=y"
+" # CONFIG_BLK_DEV_THROTTLING_LOW is not set"
+" CONFIG_BLK_WBT=y"
+"-CONFIG_BLK_WBT_MQ=y"
+" # CONFIG_BLK_CGROUP_IOLATENCY is not set"
+" # CONFIG_BLK_CGROUP_IOCOST is not set"
+" # CONFIG_BLK_CGROUP_IOPRIO is not set"
+# ...
+"Config file has unexpected changes"
+"Update config file to set changed values explicitly"
+```
+
+We see the error message `"Config file has unexpected changes... Update config file to set changed values explicitly"`. This is a common error when editing configs. CBL-Mariner's `kernel` requires implicitly enabled settings to be explicitly set. In this case, enabling only `CONFIG_BLK_WBT` is insufficient. We also need `CONFIG_BLK_WBT_MQ` to be explicitly set. The logs show that the `kernel` spec (specifically `make oldconfig`) has noted that `CONFIG_BLK_WBT_MQ` is missing. Because `CONFIG_BLK_WBT_MQ` is missing, compilation of the kernel fails.  In general, when an error of this nature occurs, the build log file for the kernel will indicate what needs to be changed.  
+
+</br>
+
+Let's fix this build error and rebuild.
+
+```bash
+# Enter kernel spec folder
+pushd ../SPECS/kernel
+
+# Update the config file with missing CONFIG_BLK_WBT_MQ=y
+sed -i '/CONFIG_BLK_WBT=y/a\CONFIG_BLK_WBT_MQ=y' config
+
+# Update the signature file
+CONFIGHASH=$(sha256sum config | awk '{print $1}')
+sed -i 's/    "config": .*/    "config": "'"$CONFIGHASH"'",/' kernel.signatures.json
+
+# Go back to toolkit
+popd
+
+# Rebuild the kernel. This will take some time. Watch the log file to see if it reaches the %build phase
+sudo make build-packages REBUILD_TOOLS=y CONFIG_FILE= PACKAGE_REBUILD_LIST="kernel"
+```
+
+If successful, you will see a message like this:
+
+```bash
+INFO[2416] ---------------------------                  
+INFO[2416] --------- Summary ---------                  
+INFO[2416] ---------------------------                  
+INFO[2416] Number of built SRPMs:             1         
+INFO[2416] Number of prebuilt SRPMs:          2         
+INFO[2416] Number of failed SRPMs:            0         
+INFO[2416] Number of blocked SRPMs:           0         
+INFO[2416] Number of unresolved dependencies: 0         
+INFO[2416] Built SRPMs:                                 
+INFO[2416] --> kernel-5.15.102.1-100.cm2.src.rpm   
+```
+
+You can also see the built RPM in `CBL-MarinerTutorials/out/RPMS/`. This RPM can be `scp`'d to a running image and installed via `sudo rpm -ihv kernel-5.15.102.1-100.cm2.rpm`.
+
+### Build an Image with a Custom Kernel
+
+We can also build an image with our new kernel. Note that even if we hadn't prebuilt the kernel in the previous steps, this step would also rebuild the kernel RPM.
+
+```bash
+# Enter the toolkit
 cd CBL-MarinerTutorials/toolkit
+
+# Clean our build environment
 sudo make clean
+
+# Make an image
```

Comment:
We talked earlier about the scenario where you just want to build the package but aren't interested to build a full custom image using the Mariner toolkit. What do you think about adding those instructions you directly sent me as well as an alternative option?

Reply:
They are currently in the "Build a Custom Kernel" but I can change the title to be more explicit to "Build a Custom Kernel RPM"